app/core/Engine.ts

```diff
@@ -1306,7 +1319,7 @@ class Engine {
 
     this.controllerMessenger.subscribe(
       AppConstants.NETWORK_STATE_CHANGE_EVENT,
-      (state: NetworkState) => {
+      async (state: NetworkState) => {
```

Comment:
Nit: It seems that this change is unnecessary

```suggestion
      (state: NetworkState) => {
```